src/commands/data/bulk/results.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { EOL } from 'node:os';
+import { writeFile } from 'node:fs/promises';
+import { SfCommand, Flags } from '@salesforce/sf-plugins-core';
+import { Messages } from '@salesforce/core';
+import ansis from 'ansis';
+import { JobInfoV2 } from '@jsforce/jsforce-node/lib/api/bulk2.js';
+
+Messages.importMessagesDirectoryFromMetaUrl(import.meta.url);
+const messages = Messages.loadMessages('@salesforce/plugin-data', 'data.bulk.results');
+
+export type DataBulkResultsResult = {
+  status: JobInfoV2['state'];
+  operation: JobInfoV2['operation'];
+  object: JobInfoV2['object'];
+  processedRecords: number;
+  successfulRecords?: number;
+  failedRecords?: number;
+  successFilePath: string;
+  failedFilePath?: string;
+  unprocessedFilePath?: string;
+};
+
+export default class DataBulkResults extends SfCommand<DataBulkResultsResult> {
+  public static readonly summary = messages.getMessage('summary');
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages.getMessages('examples');
+
+  public static readonly flags = {
+    'job-id': Flags.salesforceId({
+      summary: messages.getMessage('flags.job-id.summary'),
+      char: 'i',
+      required: true,
+      startsWith: '750',
+    }),
+    'target-org': Flags.requiredOrg(),
+    'api-version': Flags.orgApiVersion(),
+  };
+
+  public async run(): Promise<DataBulkResultsResult> {
+    const { flags } = await this.parse(DataBulkResults);
+
+    const conn = flags['target-org'].getConnection(flags['api-version']);
+
+    const job = conn.bulk2.job('ingest', {
+      id: flags['job-id'],
+    });
+
+    const jobInfo = await job.check().catch((error: Error) => {
```

Comment:
`await` and `.catch` together seems weird? We shouldn't have `.catch`

Reply:
agree. 
The `catch` is chained here to catch bad IDs early and throw a nice error (if the user pass a bulk query ID we can catch it here instead of doing:
```ts
let jobInfo = JobInfoV2 | undefined
try {
  jobInfo = await job.check()
} catch() {
...
}
```

if we wrap everything in a try/catch then we don't know from were the `The requested resource does not exist` came from.